### DISCUSSION TITLE
lesson 6 , Not compiling

### DISCUSSION THREAD

**Discussion by sivajisj** [Cyfrin/foundry-full-course-cu#352] [Q&A]
```
sivaji@DESKTOP-934I8JS:~/solidity-course/foundry-f23/foundry-simple-storage-f23$ forge create SimpleStorage --rpc-url  http://127.0.0.1:7545 --interactive
[⠊] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://127.0.0.1:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
- ```   
- getting this error while compiling please help me

**Comment by shikhar229169** [comment]
Ensure that your anvil chain is running in another terminal.
To run an anvil chain, type in shell
```
anvil
```

**Reply by sivajisj** [reply]
Bro I am using ganache , the created anvil network rpc link also changed but , while asking private key my terminal not taking input
   
   I don't understand what's the problem

**Reply by 0xepicfailll** [reply]
Help pls, same situation. How did u fix that?

**Comment by sivajisj** [comment]
![p](https://github.com/Cyfrin/foundry-full-course-f23/assets/93958018/2c29933a-ea95-49c1-a983-99f1565d1bdf)

**Reply by sivajisj** [reply]
but not taking private key , i tried manually and copy/paste but it is freezed
  please help me anyone

**Reply by shikhar229169** [reply]
It is pasted for sure, but remains hidden.
So just paste it one time and press enter.

**Reply by sivajisj** [reply]
Thank you brother, it's working fine ☺️

**Reply by vectordotsats** [reply]
Thanks Brother, I literally just followed these steps and my problems were perfectly solved.

**Comment by ShawnFromTexas** [comment]
Hey mate, if issue is not sorted go to SETTINGS in Ganache at top right >> then to SERVER >> change HOSTNAME to the WSL version and update/copy your new "http://172.25.224.1:7545" into code. 

Let me know if it works.
